src/Views/HardwareView.vala

```diff
@@ -280,6 +284,69 @@ public class About.HardwareView : Gtk.Grid {
 
         return 0;
     }
+
+    private string get_storage_type () {
+        string partition_name = get_partition_name ();
+        string disk_name = get_disk_name (partition_name);
+        string path = "/sys/block/%s/queue/rotational".printf (disk_name);
+        string storage = "";
+        try {
+            string contents;
+            FileUtils.get_contents (path, out contents);
+            if (int.parse (contents) == 0) {
+                if (disk_name.has_prefix ("nvme")) {
+                    storage = _("storage (NVMe SSD)");
```

Comment:
Sorry for going in circles here :sweat_smile: But at this point, wouldn't it be easier to have individual strings like "%s storage (NVMe SSD)", etc. for each storage type?

Reply:
Fixed!